checks/__init__.py

```diff
@@ -535,8 +552,61 @@ def get_warnings(self):
         self.warnings = []
         return warnings
 
+    @staticmethod
+    def _get_statistic_name_from_method(method_name):
+        return method_name[4:] if method_name.startswith('get_') else method_name
+
+    @staticmethod
+    def _collect_stats(methods=None, log=None):
```

Comment:
You should find a better name:
https://github.com/DataDog/dd-agent/blob/master/checks.d/postgres.py#L351
https://github.com/DataDog/dd-agent/blob/master/checks.d/pgbouncer.py#L69

What about 
```python
_collect_internal_stats
```
?